File: 卡码/代码随想录/1/07-回溯/Js/01-组合/LeetCode_0017.js
//17. 电话号码的字母组合
const letterCombinations = function(digits) {
  const nums=["","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"];
  const res=new Array();
  const backtracking=function(digits,index,str){
    if(index==digits.length){
      res.push(str);
      return;
    }
    const digit=digits.charAt(index);
    const letter=nums[digit-'0'];
    for(let i=0;i<letter.length;i++){
      backtracking(digits,index+1,str+letter[i]);
    }
  }
  if(digits.length!=0) backtracking(digits,0,"");
  return res;
};

File: 卡码/代码随想录/1/07-回溯/Go/01-组合/LeetCode_0017.go
//17. 电话号码的字母组合
package main

func letterCombinations(digits string) []string {
	var nums=[]string{"","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"}
	res:=make([]string,0)
	var backtracking func(digits,str string,index int)
	backtracking=func(digits,str string,index int){
		if index==len(digits) {
			res=append(res,str)
			return
		}
		digit:=digits[index]-'0'
		letter:=nums[digit]
		for i:=0;i<len(letter);i++ {
			backtracking(digits,str+string(letter[i]),index+1)
		}
	}
	if len(digits)!=0 {
		backtracking(digits,"",0)
	}
	return res
}
